crates/bitwarden/src/client/client.rs

```diff
@@ -91,12 +122,80 @@ impl Client {
             .build()
             .unwrap();
 
+        let mut temp_token: Option<String> = None;
```

Comment:
I think what we're doing here would be clearer if we mostly kept this function the same as it was before, we created the Client instance, and then we did something like: 
```rust
if let Some(client_state) = client_state {
    client.set_login_method(...);
    client.set_tokens(...);
    client.initialize_crypto_single_key(...)
}

return client
```

Reply:
Good call - refactored